storage/database/db_manager_test.go

```diff
@@ -351,6 +352,73 @@ func TestDBManager_Block(t *testing.T) {
 	}
 }
 
+func TestDBManager_BadBlock(t *testing.T) {
+	header := &types.Header{Number: big.NewInt(int64(num1))}
+	headertwo := &types.Header{Number: big.NewInt(int64(num2))}
+	for _, dbm := range dbManagers {
+		// block #1 test
+		block := types.NewBlockWithHeader(header)
+
+		if entry := dbm.ReadBadBlock(block.Hash()); entry != nil {
+			t.Fatalf("Non existance block returned, %v", entry)
+		}
+		dbm.WriteBadBlock(block)
+		if entry := dbm.ReadBadBlock(block.Hash()); entry == nil {
+			t.Fatalf("Existing bad block didn't returned, %v", entry)
+		} else if entry.Hash() != block.Hash() {
+			t.Fatalf("retrived block mismatching, have %v, want %v", entry, block)
+		}
+		if badblocks, _ := dbm.ReadAllBadBlocks(); len(badblocks) != 1 {
+			for _, b := range badblocks {
+				t.Log(b)
+			}
+			t.Fatalf("bad blocks length mismatching, have %d, want %d", len(badblocks), 1)
+
+		}
+
+		// block #2 test
+		blocktwo := types.NewBlockWithHeader(headertwo)
+		dbm.WriteBadBlock(blocktwo)
+		if entry := dbm.ReadBadBlock(blocktwo.Hash()); entry == nil {
+			t.Fatalf("Existing bad block didn't returned, %v", entry)
+		} else if entry.Hash() != blocktwo.Hash() {
+			t.Fatalf("retrived block mismatching, have %v, want %v", entry, block)
+		}
+
+		// block #1 insert again
+		dbm.WriteBadBlock(block)
+		badBlocks, _ := dbm.ReadAllBadBlocks()
+		if len(badBlocks) != 2 {
+			t.Fatalf("bad block db len mismatching, have %d, want %d", len(badBlocks), 2)
+		}
+
+		// Write a bunch of bad blocks, all the blocks are should sorted
+		// in reverse order. The extra blocks should be truncated.
+		for _, n := range rand.Perm(150) {
```

Comment:
```suggestion
		for _, n := range rand.Perm(110) {
```

Reply:
BadgerDB fails to open a new database in another TC if this test writes some amount of data. I don't know the exact reason for it yet, but I don't this the reason is related to this PR. So, I just reduce the size of stored data in this TC. 
It would be better if you investigate more why BadgerDB failed with large data after merging this PR.

Reply:
I'll make an issue. Thanks.